src/tigerasi/tiger_controller.py

```diff
@@ -888,29 +964,39 @@ def start_array_scan(self, wait: bool = True):
         """
         # Use the previously specified card address.
         if self._array_scan_card_addr is None:
-            raise RuntimeError("Cannot infer the card address for which to "
-                               "apply the sttings. setup_array_scan must be "
-                               "run first.")
+            raise RuntimeError(
+                "Cannot infer the card address for which to "
+                "apply the sttings. setup_array_scan must be "
+                "run first."
+            )
         card_address = self._array_scan_card_addr
         # Clear card address for which the array-scan settings were specified.
         self._array_scan_card_addr = None
-        self._set_cmd_args_and_kwds(Cmds.ARRAY, card_address=card_address,
-                                    wait=wait)
+        self._set_cmd_args_and_kwds(Cmds.ARRAY, card_address=card_address, wait=wait)
 
-    def reset_ring_buffer(self, wait: bool = True):
+    def reset_ring_buffer(self, axis: str = None, wait: bool = True):
         """Clear the ring buffer contents."""
-        self._clear_ring_buffer(wait=wait)
-        self._rb_axes = []
+        if axis:
+            self._clear_ring_buffer(axis=axis, wait=wait)
+        else:
+            self._clear_ring_buffer(wait=wait)
+            self._rb_axes = []
 
```

Comment:
Let's consolidate these two functions into just one since they're only used here, and `reset_ring_buffer` appears to be a wrapper for `_clear_ring_buffer` without really adding anything.